### PR TITLE
Fix: non node images fail with no error message

### PR DESCRIPTION
**Pull request by TylerBarnes**
### Bug Fixes & internal changes

- Since we unified the code for fetching media item nodes and file nodes in html fields, a regression was introduced where our http error handling was assuming it was still always for a media item node. This oversight obscured errors for html image file sourcing.
- Sort actions by modified date instead of date since there's now only 1 action for each node which is just modified instead of being created new.
- The last sourced time was moved from the end of sourcing data to the beginning. The reason for this is that if updates happen during node sourcing, they could previously be potentially missed. Now they'll be picked up.
- Static file link replacement could previously be triggered on MediaItem nodes. This was problematic because the sourceUrl/mediaItemUrl field would be overwritten with a static file link. When the plugin later went to use these fields to create the localFile field, it was no longer a valid url for the WP site and the build would crash. Static file link replacement is now disabled for MediaItem nodes to prevent this from happening.
- Removed dead code which was keeping track of actions and removing duplicates. We now only ever store 1 action per node so duplicates aren't possible anymore.

### New Features

- Because 404's can block many sites from completing production builds, we've added an option to allow 404s to not fail production builds.

```
{
  options: {
    production: {
      allow404Images: true
    }
  }
}
```

### CODE REVIEW

**Comment by TylerBarnes** [comment]
Closes #375, #206, and #384

File: plugin/src/steps/source-nodes/update-nodes/content-update-interval.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ import { LAST_COMPLETED_SOURCE_TIME } from "../../../constants"
  */
 const checkForNodeUpdates = async ({ cache, emitter }) => {
   // get the last sourced time
-  const since = await cache.get(LAST_COMPLETED_SOURCE_TIME)
+  const lastCompletedSourceTime = await cache.get(LAST_COMPLETED_SOURCE_TIME)
+  const since = lastCompletedSourceTime - 500
```
Comment (smthomas) [inline plugin/src/steps/source-nodes/update-nodes/content-update-interval.js:15]: Where does the 500 value come from here?

Comment (TylerBarnes) [reply plugin/src/steps/source-nodes/update-nodes/content-update-interval.js:15]: It's somewhat arbitrary - often we were missing node updates by a few milliseconds during development. It has no effect on production builds but makes development content updates reliable. We actually had this in here for a really long time and I removed it before the holidays when Jason did his big refactor. I thought the refactor had removed the need for this but it's still an issue.

Comment (TylerBarnes) [reply plugin/src/steps/source-nodes/update-nodes/content-update-interval.js:15]: If we switched from using a timestamp to a cursor this wouldn't be an issue anymore, but that's too big of a change right now :)

Comment (TylerBarnes) [reply plugin/src/steps/source-nodes/update-nodes/content-update-interval.js:15]: This is the problem -> https://github.com/gatsbyjs/gatsby-source-wordpress-experimental/issues/386

**Review by smthomas — APPROVED** [review]

**Review by jasonbahl — APPROVED** [review]

**Comment by TylerBarnes** [comment]
Published in `gatsby-source-wordpress-experimental@6.1.0`

**Pull request merged**